OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -679,6 +679,20 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				// Fixed position targeting and added TargetableLocations, while removing ITargetablePositions from Building
+				if (engineVersion < 20170514)
+				{
+					var building = node.Value.Nodes.FirstOrDefault(n => n.Key == "Building");
+					if (building != null)
+					{
+						Console.WriteLine("The Building trait no longer provides per-cell targetable positions,");
+						Console.WriteLine("and weapons no longer unconditionally target actor CenterPosition.");
+						Console.WriteLine("You may need to manually set custom target locations or update building hit shapes.");
```

Comment:
Just pointing out for others that the changes in this PR mean that buildings in ra/cnc/ts will now be targeted only at their CenterPosition.  This will reintroduce several old bugs relating to attack ranges.  By merging this PR we are committing to running the hitshape changes through the other mods, which will be great for the long term but painful for balancing in the short term.